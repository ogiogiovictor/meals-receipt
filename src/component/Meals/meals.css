.meals-bg {
    background-color: var(--bgrey);
    overflow-x: hidden;
    padding: 20px;
    height: fit-content;
}

.imageSize {
    width: 100%;

}
.title {
    font-size: 1.1rem;
    padding:10px;
}

.listags {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 10px;
}

.listags li {
    list-style: none;
    font-size: 0.8rem;
    color: var(--bgcontact);
}

.buttonLight{
    width: 100%;
    background-color: #ec4c8b;
    border: none;
    color: white;
    padding: 5px 2px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    cursor: pointer;
}

.meal-header {
    display: flex;
    flex-direction: row;
}

.meal-header h1{
    flex: 1;
}

.meal-header input{
    flex: 3;
    border: 1px solid #ccc;
    border-radius: 5px;
    padding: 7px;
    margin-right: 30px;

}
.meal-header select {
    flex: 2;
    border: 1px solid #ccc;
    margin-right: 7px;
}

