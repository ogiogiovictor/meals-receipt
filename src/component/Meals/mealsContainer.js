import React from 'react';
import SingleMeal from './singleMeal';

const mealsContainer = ({ meals }) => (
  <div className="grid-container">
    {meals.map((ml) => (

      <SingleMeal
        key={ml.id}
        id={ml.id}
        meal_name={ml.meal_name}
        category={ml.category}
        mealInstruction={ml.mealInstruction}
        mealImage={ml.mealImage}
        youtube={ml.youtube}
        mealSource={ml.mealSource}
        area={ml.area}
      />

    ))}

  </div>
);

export default mealsContainer;
