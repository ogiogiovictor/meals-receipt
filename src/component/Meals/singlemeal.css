.meal-details {
  padding-bottom: 10px;
}

.details {
  display: flex;
  flex-direction: row;
  padding-bottom: 20px;
}

.meal-image {
  border: 1px solid rgb(234, 234, 234);
  border-radius: 5px;
  width: 95%;
}

.mealitems {
  width: 90%;
}

.mealitems ul {
  padding-left: 0;
  list-style-type: none;
}

.mealitems ul li {
  border: 1px solid #ededed;
  border-radius: 5px;
  background-color: #fff;
  display: inline-block;
  color: #2d2013;
  font-size: 12px;
  font-weight: bold;
  padding: 5px;
  margin: 0 5px 5px 0;
}

.instructions {
  font-family: 'Roboto', sans-serif;
  text-align: justify;
  line-height: 1.6rem;
}
