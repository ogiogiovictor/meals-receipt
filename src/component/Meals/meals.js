/* eslint-disable */
import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMeals } from '../../redux/meals/mealslice';
import { getCategory } from '../../redux/category/categorySlice';
import MealContainer from './mealsContainer';

import './meals.css';

const Meals = () => {
  const allMeals = useSelector((state) => state.meals);
  const [filtered, setFiltered] = useState([]);
  const category = useSelector((state) => state.category);
  const [cat, setCat] = useState('');

  const dispatch = useDispatch();

  useEffect(() => {
    if (!allMeals.length) {
      dispatch(getMeals());
    }
  }, [allMeals.length]);

  useEffect(() => {
    if (!category.length) {
      dispatch(getCategory());
    }
    if (cat.length) {
      setFiltered(allMeals.filter((meal) => meal.category === cat));
    }
  }, [category.length]);

  const filteredMenu = allMeals.filter((meal) => meal.meal_name.toLowerCase().includes(filtered));

  const filteredCategory = allMeals.filter((meal) => meal.category === cat);

  // const handleFilter = filtered.length > 0 ? filteredMenu : filteredCategory;
  const handleFilter = cat.length && cat !== 'All' ? filteredCategory : filteredMenu;

  return (

    <div className="meals-bg">
      <div className="meal-header">
        <h1>Meals</h1>
        <input type="search" placeholder="Search Meal" value={filtered} onChange={(e) => setFiltered(e.target.value)} />
        <select value={cat} onChange={(e) => setCat(e.target.value)}>
          <option value="All">All Meals</option>
          {
                category.map((cat) => (<option key={cat.id} value={cat.meal_name}>{cat.meal_name}</option>))

                }
        </select>
      </div>

      {/* <p className="home-results">{filtered.length ? `Showing ${filtered.length} results` : `Showing ${allMeals.length} results`}</p> */}
      {

        handleFilter.length ? <MealContainer meals={handleFilter} /> : <MealContainer meals={allMeals} />
        }
    </div>
  );
};

export default Meals;
