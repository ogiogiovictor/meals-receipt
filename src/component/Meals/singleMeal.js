import React from 'react';
import {Link } from 'react-router-dom';
import './meals.css';

const SingleMeal = ({
    id, meal_name, category, mealInstruction, mealImage, youtube,  mealSource, area
  }) => {
    return (
        <div className="grid-flex">
            <Link to={`/meal/${id}`} >
            <div className="g-card-body">
                <img src={mealImage} className="imageSize"/>
                <p className="title">{ meal_name }</p><hr/>
                <ul className="listags">
                    <li>{ category }</li>
                    <li>{ area }</li>
                </ul>
                <div className="buttonLight">View Instructions</div>
            </div>
            </Link>
        </div>
    )
}

export default SingleMeal;