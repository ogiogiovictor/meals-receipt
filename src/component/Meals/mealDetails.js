import React from 'react';
import { useParams, Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { ImReply } from 'react-icons/im';
import SingleMealDetails from './singleMealdetails';
import './singlemeal.css';

const MealDetails = () => {
  // const { id } = useParams();
  const params = useParams();
  const meals = useSelector((state) => state.meals);
  const findmeals = meals.find((meal) => meal.id === params.id);

  return (
    <div className="meals-bg">
      <h1 className="meal-details">
        Meal Details
        <Link to="/" className="pushTop">
          {' '}
          <ImReply size={30} />
        </Link>
      </h1>

      <SingleMealDetails
        key={findmeals.id}
        id={findmeals.id}
        meal_name={findmeals.meal_name}
        category={findmeals.category}
        mealInstruction={findmeals.mealInstruction}
        mealImage={findmeals.mealImage}
        youtube={findmeals.youtube}
        mealSource={findmeals.mealSource}
        area={findmeals.area}
        ingredients={findmeals.ingredients}
        soup={findmeals.soup}
      />
    </div>
  );
};

export default MealDetails;
