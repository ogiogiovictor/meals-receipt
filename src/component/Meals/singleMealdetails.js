/* eslint-disable */
import React, { Fragment } from 'react';
import ReactPlayer from 'react-player';
import { PropTypes } from 'prop-types';
import './singlemeal.css';

const SingleMealDetails = ({
  id, meal_name, category, area, mealInstruction, ingredients,
  mealImage, mealSource, soup, youtube,
}) => (
  <>
    <div className="details">
      <div className="image">
        <img src={mealImage} className="meal-image" alt={id} />

      </div>
      <div className="mealitems">
        <div>
          Meal Name:
          <b>
            {' '}
            {meal_name}
            {' '}
          </b>
        </div>
        <div>
          Category:
          <b>
            {' '}
            {category}
            {' '}
          </b>
        </div>
        <div>
          Country:
          <b>
            {' '}
            {area}
            {' '}
          </b>
        </div>
        <hr />
        <h4>Meal Instruction</h4>
        <div className="instructions">{mealInstruction}</div>
        <h4>Meal Ingredients</h4>
        <hr />
        <ul>
          {
                     ingredients.map((ing) => (
                       <li>{ing}</li>
                     ))

                }
        </ul>

        <span className="moreLink">
          More Info Visit -
          {soup}
        </span>
        {' '}
        <a href={mealSource} target="_blank" rel="noreferrer">View Source</a>

      </div>
    </div>

    <div className="player-wrapper">
      <ReactPlayer
        url={youtube}
        className="react-player"
        playing
      />
    </div>
  </>
);

SingleMealDetails.propTypes = {
  id: PropTypes.string.isRequired,
  meal_name: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  mealInstruction: PropTypes.bool.isRequired,
  mealImage: PropTypes.string.isRequired,
  youtube: PropTypes.string.isRequired,
  mealSource: PropTypes.string.isRequired,
  area: PropTypes.string.isRequired,
  ingredients: PropTypes.string.isRequired,
  soup: PropTypes.string.isRequired,
};

export default SingleMealDetails;
