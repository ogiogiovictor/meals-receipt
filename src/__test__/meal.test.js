import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Meals from '../component/Meals/Meals';

const TestRenderer = require('react-test-renderer');

it('Renders meals correctly', () => {
  const tree = TestRenderer
    .create(
      <Provider store={store}>
        <Meals />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});