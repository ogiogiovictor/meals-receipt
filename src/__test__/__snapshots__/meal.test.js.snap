// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders meals correctly 1`] = `
<div
  className="meals-bg"
>
  <div
    className="meal-header"
  >
    <h1>
      Meals
    </h1>
    <input
      onChange={[Function]}
      placeholder="Search Meal"
      type="search"
      value={Array []}
    />
    <select
      onChange={[Function]}
      value=""
    >
      <option
        value="All"
      >
        All Meals
      </option>
    </select>
  </div>
  <div
    className="grid-container"
  />
</div>
`;
